libs/velo-external-db-core/src/utils/schema_utils.ts

```diff
@@ -0,0 +1,123 @@
+            return 'text'
+        case FieldType.number:
+            return 'number'
+        case FieldType.datetime:
+            return 'datetime'
+        case FieldType.boolean:
+            return 'boolean'
+        case FieldType.object:
+            return 'object'
+
+        case FieldType.singleReference:
+        case FieldType.multiReference:
+        default:
+            // TODO: throw specific error
+            throw new Error(`Unsupported field type: ${fieldEnum}`)
+    }
+}
+
+export const subtypeToFieldType = (fieldEnum: number): string => {
+    switch (fieldEnum) {
+        case FieldType.text:
+        case FieldType.longText:
+            return 'string'
+        case FieldType.number:
+            return 'int'
+        case FieldType.datetime:
+            return 'datetime'
+        case FieldType.boolean:
+            return ''
+        case FieldType.object:
+            return ''
+
+        case FieldType.singleReference:
+        case FieldType.multiReference:
+        default:
+            // TODO: throw specific error
+            throw new Error(`There is no subtype for this type: ${fieldEnum}`)
+    }
+
+}
+
+export const convertQueryOperatorsToEnum = (queryOperator: string): QueryOperator => {
+    switch (queryOperator) {
+        case eq:
+            return QueryOperator.eq
+        case lt:
+            return QueryOperator.lt
+        case gt:
+            return QueryOperator.gt
+        case ne:
+            return QueryOperator.ne
+        case lte:
+            return QueryOperator.lte
+        case gte:
+            return QueryOperator.gte
+        case string_begins:
+            return QueryOperator.startsWith
+        case string_ends:
+            return QueryOperator.endsWith
+        case string_contains:
+            return QueryOperator.contains
+        case include:
+            return QueryOperator.hasSome
+        // case 'hasAll':
+            // return QueryOperator.hasAll
+        // case 'exists':
+            // return QueryOperator.exists
+        // case 'urlized':
+            // return QueryOperator.urlized
+        default:
+            throw new Error(`${queryOperator} - Unsupported query operator`)
+    }    
+} 
+
+export const convertQueriesToQueryOperatorsEnum = (queryOperators: string[]): QueryOperator[] => {
+    return queryOperators.map(convertQueryOperatorsToEnum)
+}
+
+export const convertResponseFieldToWixFormat = (fields: ResponseField[]): Field[] => {
+    return fields.map(field => {
+        return {
+            key: field.field,
+            type: convertFieldTypeToEnum(field.type)
+        }
+    })
+}
+
+export const convertWixFormatFieldToInputFields = (field: Field): InputField => ({
+    name: field.key,
+    type: convertEnumToFieldType(field.type),
+    subtype: subtypeToFieldType(field.type)
+})
+
+export const convertWixFormatFieldsToInputFields = (fields: Field[]): InputField[] => fields.map(convertWixFormatFieldToInputFields)
```

Comment:
I think it would be more understandable - `spiFieldsToDomainFields`, or if there's a difference between input field and output Field from some reason so  - `spiFieldsToDomainInputFields`